src/AutoMapper/TypeExtensions.cs

```diff
@@ -36,18 +36,35 @@ public static Type CreateType(this TypeBuilder type)
 
         public static IEnumerable<MemberInfo> GetAllMembers(this Type type)
         {
-            while (true)
+            if (type.IsInterface())
```

Comment:
What do you mean by that?
I just tried to keep the amount of changes as little as possible.